test/lib/jdk/test/lib/process/ProcessTools.java

```diff
@@ -823,7 +824,7 @@ static class MainThreadGroup extends ThreadGroup {
         }
 
         public void uncaughtException(Thread t, Throwable e) {
-            if (e instanceof ThreadDeath) {
+            if (e instanceof Error && t.getState() == State.TERMINATED) {
```

Comment:
Dropping the check for ThreadDeath is fine but checking the thread state is puzzling. When a Thread completes with an uncaught exception/error then the UHE is called from the thread before it terminates so it's state won't be TERMINATED. Is there more going on here that the UHE is called with an already terminated Thread?

Reply:
ok this is a strange function.  What does line 833 do?

Reply:
I assumed the `ThreadDeath` was there to catch `Thread.stop()` calls and just ignore them. So, I wanted to keep this functionality.
Can we remove using `ThreadDeath` relying on that `Thread.stop()` is also deprecated?

Reply:
Removed.